### PR TITLE
SwingSet ESLint JSDoc

### PR DESCRIPTION
**Pull request by michaelfig**
This is required to get our JSDoc in SwingSet up to Zoe standards.

Note that many of the types are `{*}` which means, "any" but indicates that I didn't have the wherewithal to make a more specific type.

### CODE REVIEW

File: packages/SwingSet/src/controller.js [hunk 1]
```diff
@@ -52,29 +52,38 @@ const KNOWN_CREATION_OPTIONS = harden([
...
  * the `bootstrap` property names the vat that should be used as the bootstrap vat.  Although a swingset
  * configuration can designate any vat as its bootstrap vat, `loadBasedir` will always look for a file named
  * 'bootstrap.js' and use that (note that if there is no 'bootstrap.js', there will be no bootstrap vat).
+ */
+
+/**
+ * @typedef {Object} SwingSetConfig a swingset config object
+ * @property {string} bootstrap
+ * @property {SwingSetConfigDescriptor} [vats]
+ * @property {SwingSetConfigDescriptor} [bundles]
```
Comment (FUDCo) [inline packages/SwingSet/src/controller.js:73]: Also need devices and (I think) one or two other properties.

**Review by FUDCo — CHANGES_REQUESTED** [review]
Wow!  Very impressive.

I flagged a few bits of explanation and comment formatting got lost or mangled in the editing, which should be patched up before this lands.

File: packages/SwingSet/src/capdata.js [hunk 2]
```diff
@@ -5,12 +5,10 @@ import { assert, details } from '@agoric/assert';
  * actually is.  A capdata object should have a .body property that's a string
  * and a .slots property that's an array of strings.
  *
- * @param capdata  The object to be tested
- *
- * @throws Error if, upon inspection, the parameter does not satisfy the above
+ * @param {any} capdata  The object to be tested
```
[the remark this answers is not in the document]
Comment (warner) [reply packages/SwingSet/src/capdata.js:9]: I wouldn't know how to write the type spec, but for the record it's `{ body, slots }`, where `body` is a string (the output of a JSON.stringify, specifically `marshal`), and `slots` is an array of slot specifiers, all of which are strings. Various places have kernel-slots or vat-slots, which refines the PREFIX of `${prefix}${joiner}${number}`.

File: packages/SwingSet/demo/encouragementBot/bootstrap.js [hunk 3]
```diff
@@ -1,4 +1,3 @@
-/* global harden */
 import { E } from '@agoric/eventual-send';
```
Comment (warner) [inline packages/SwingSet/demo/encouragementBot/bootstrap.js:1]: Hm, pity, my main editor (`js2-mode` on emacs) knows about the `/* global` comment but not about external `.eslintrc` files. I'll look for a different way to teach it about globals.

I assume eslint is now complaining about redundant `/* global */` declarations?

File: packages/SwingSet/src/controller.js (hunk 1 above)
Comment (warner) [reply packages/SwingSet/src/controller.js:73]: I did a quick scan, I think adding `@property {any} devices` is probably sufficient.

File: packages/SwingSet/src/hostStorage.js [hunk 4]
```diff
@@ -58,29 +56,37 @@ export function buildHostDBInMemory(storage) {
...
+  /**
+   * @typedef {Object} DeleteOperation
+   * @property {'delete'} op
+   * @property {string} key
+   */
+
   /**
    * Make an ordered set of changes to the state that is stored.  The changes
    * are described by a series of change description objects, each of which
-   * describes a single change.  There are currently two forms:
```
[the remark this answers is not in the document]
Comment (warner) [reply packages/SwingSet/src/hostStorage.js]: I think the informal lines removed were translated into the `SetOperation`/`DeleteOperation` typedefs above, yeah? Maybe just add "`, described above`"?

File: packages/SwingSet/src/kernel/deviceManager.js [hunk 5]
```diff
@@ -51,16 +50,19 @@ export default function makeDeviceManager(
...
    * Invoke a method on a device node.
    *
-   * @param target  Kernel slot designating the device node that is the target
-   *    of the invocation
-   * @param method  A string naming the method to be invoked
-   * @param args  A capdata object containing the arguments to the invocation
-   *
-   * @return a VatInvocationResults object: ['ok', capdata]
-   *
+   * @param {[string, string, CapData]} deviceInvocation
```
Comment (warner) [inline packages/SwingSet/src/kernel/deviceManager.js]: The descriptions of the three components of `deviceInvocation` need to be retained somewhere, just adding `[target, method, args]` to the comment would be sufficient.

**Review by warner — CHANGES_REQUESTED** [review]
I'll second Chip's retain-docs requests, and added a few comments. The other swingset changes look good to me. I've got no expert opinion on the zoe/ertp/etc changes, but they look ok to me.

File: packages/SwingSet/demo/encouragementBot/bootstrap.js (hunk 3 above)
Comment (michaelfig) [reply packages/SwingSet/demo/encouragementBot/bootstrap.js:1]: > I assume eslint is now complaining about redundant /* global */ declarations?

Yep.

File: packages/SwingSet/src/controller.js [hunk 6]
```diff
@@ -52,29 +52,38 @@ const KNOWN_CREATION_OPTIONS = harden([
...
 
 /**
- * Scan a directory for files defining the vats to bootstrap for a swingset, and
- * produce a swingset config object for what was found there.  Looks for files
- * with names of the pattern `vat-NAME.js` as well as a file named
- * 'bootstrap.js'.
- *
- * @param basedir  The directory to scan
- *
- * @return a swingset config object: {
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply packages/SwingSet/src/controller.js:62]: Okay, will do!

File: packages/SwingSet/src/hostStorage.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (michaelfig) [reply packages/SwingSet/src/hostStorage.js]: The IDE user reading this does not have a notion of "above".  I've restored the old documentation.

File: packages/SwingSet/src/kernel/deviceManager.js (hunk 5 above)
Comment (michaelfig) [reply packages/SwingSet/src/kernel/deviceManager.js]: I've done the moral equivalent of this.

**Review by FUDCo — APPROVED** [review]
Stupendous!

**Pull request merged**